core/common/src/internal/format/parser/ParserOperation.kt

```diff
@@ -158,95 +158,119 @@ internal class TimeZoneParserOperation<Output>(
             START,
             AFTER_PREFIX,
             AFTER_SIGN,
+            AFTER_INIT_SIGN,
             AFTER_HOUR,
+            AFTER_INIT_HOUR,
             AFTER_MINUTE,
             AFTER_COLON_MINUTE,
-            END,
-            INVALID
+            IN_PART,
+            AFTER_SLASH,
+            END
         }
 
-        private fun validateTimezone(input: CharSequence, startIndex: Int): Int {
+        private fun validateTimeZone(input: CharSequence, startIndex: Int): Int {
             var index = startIndex
-            var lastValidIndex = startIndex
 
             fun validatePrefix(validValues: List<String>): Boolean =
-                validValues.firstOrNull { input.startsWith(it) }?.let {
+                validValues.firstOrNull { input.startsWith(it, startIndex) }?.let {
```

Comment:
```suggestion
                validValues.firstOrNull { input.startsWith(it, index) }?.let {
```
so that this function could theoretically be used in later parts.